src/main/java/org/folio/inventory/dataimport/handlers/actions/UpdateItemEventHandler.java

```diff
@@ -68,29 +75,44 @@ public UpdateItemEventHandler(Storage storage) {
         return future;
       }
 
+      JsonObject itemAsJson = new JsonObject(payloadContext.get(ITEM.value()));
+      String oldItemStatus = itemAsJson.getJsonObject(STATUS_KEY).getString("name");
       preparePayloadForMappingManager(dataImportEventPayload);
       MappingManager.map(dataImportEventPayload);
-      JsonObject itemAsJson = new JsonObject(payloadContext.get(ITEM.value()));
-      itemAsJson = itemAsJson.containsKey(ITEM_PATH_FIELD) ? itemAsJson.getJsonObject(ITEM_PATH_FIELD) : itemAsJson;
+      JsonObject mappedItemAsJson = new JsonObject(payloadContext.get(ITEM.value()));
+      mappedItemAsJson = mappedItemAsJson.containsKey(ITEM_PATH_FIELD) ? mappedItemAsJson.getJsonObject(ITEM_PATH_FIELD) : mappedItemAsJson;
 
-      List<String> errors = validateItem(itemAsJson, requiredFields);
+      List<String> errors = validateItem(mappedItemAsJson, requiredFields);
       if (!errors.isEmpty()) {
         String msg = format("Mapped Item is invalid: %s", errors.toString());
         LOG.error(msg);
         future.completeExceptionally(new EventProcessingException(msg));
         return future;
       }
 
+      String newItemStatus = mappedItemAsJson.getJsonObject(STATUS_KEY).getString("name");
+      boolean statusWasUpdated = !oldItemStatus.equals(newItemStatus);
+      boolean isOldStatusProtected = PROTECTED_STATUSES_FROM_UPDATE.contains(oldItemStatus);
+      if(statusWasUpdated && isOldStatusProtected) {
+        mappedItemAsJson.getJsonObject(STATUS_KEY).put("name", oldItemStatus);
+      }
```

Comment:
I think it will be better to extract this logic(regarding Item`s statuses) to the separate method.